structured_external_storage_all/models/ir_attachment.py

```diff
@@ -1,8 +1,48 @@
 # -*- coding: utf-8 -*-
 # © 2018 Sunflower IT (http://sunflowerweb.nl)
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
-from odoo import models, fields, api, _
+import logging
+import functools
+import datetime
+import dateutil.relativedelta as relativedelta
+from odoo import api, models, tools
 from odoo.tools.safe_eval import safe_eval
+from werkzeug import urls
+
+_logger = logging.getLogger(__name__)
+try:
+    from jinja2.sandbox import SandboxedEnvironment
+    mako_template_env = SandboxedEnvironment(
+        block_start_string="<%",
+        block_end_string="%>",
+        variable_start_string="${",
+        variable_end_string="}",
+        comment_start_string="<%doc>",
+        comment_end_string="</%doc>",
+        line_statement_prefix="%",
+        line_comment_prefix="##",
+        trim_blocks=True,               # do not output newline after blocks
+        autoescape=True,                # XML/HTML automatic escaping
+    )
+    mako_template_env.globals.update({
+        'str': str,
+        'quote': urls.url_quote,
+        'urlencode': urls.url_encode,
+        'datetime': datetime,
+        'len': len,
+        'abs': abs,
+        'min': min,
+        'max': max,
+        'sum': sum,
+        'filter': filter,
+        'reduce': functools.reduce,
+        'map': map,
+        'round': round,
+
+        'relativedelta': lambda *a, **kw: relativedelta.relativedelta(*a, **kw)
+    })
+except ImportError:
+    _logger.warning("jinja2 not available, templating features will not work!")
```

Comment:
OK i don't think we should be as careful as this, because even Odoo 11 itself [happily imports jinja2](https://github.com/OCA/OCB/blob/11.0/addons/web/controllers/main.py#L13), expecting it to be there. So just import.